app/views/settings/index.html.erb

```diff
@@ -139,6 +139,25 @@
       </div>
     <% end %>
 
+    <% if Keybase.enabled? %>
+      <div class="boxline">
+        <%= f.label :kb_username,
+          raw("<a href=\"https://keybase.io/\">Keybase</a>:"),
+          :class => "required" %>
+        <span>
+        <% if @edit_user.keybase_signatures.present? && Keybase.verify(@edit_user.keybase_signatures[0].kb_username, @edit_user.keybase_signatures[0].sig_hash, @edit_user.username) %>
```

Comment:
Remove this. We don't want to hit an API from a view. If this is a check that has to happen before we're sure their Keybase identity is verified, put it before storing those credentials.

Reply:
this check should be done every time we provide the signature externally, where should it go?

Reply:
I don't see that in [their doc](https://keybase.io/docs/proof_integration_guide#3-linking-user-profiles) and I don't want to block our page load on a call to their service. If they were down or degraded we'd break. User profiles are some of our [highest-traffic pages](https://lobste.rs/s/cqnzl5/lobste_rs_access_pattern_statistics_for#c_lpzgbh).

I hope this isn't a dealbreaker requirement from Keybase.

Reply:
tricky thing here. we can't actually check that the proof is live in keybase before saving it in lobsters. keybase needs to see the proof being served from lobsters so it can report back to you that it's live.

Reply:
Keybase can't see `/settings` without authenticating as that user. Sounds like the PR will need to add an endpoint to serve back the proof.

Reply:
keybase currently is configured to check /u/name.json

Reply:
confirmed. keybase is hitting, e.g. `lobste.rs/u/pushcx.json`, and looking at the `"keybase_signatures"` field in there. that should all be unauthed.